walkdir.py

```diff
@@ -202,11 +202,32 @@ def onloop(dirpath):
                     continue
         yield dir_entry
 
+
+def handle_dirsymlink_as_file(walk_iter):
```

Comment:
To convey that handling of all directory symlinks is affected, and for consistency with `followlinks`, we should pluralise this: `handle_dirsymlinks_as_files`.